public/main.js

```diff
@@ -0,0 +1,497 @@
+let playerArray = [];
+let turnCount = 0;
+let trueCount = 0;
+let blockArray = [];
+let currentBox;
+let playerXName = 'X';
+let playerOName = 'O';
+let playerName;
+let player;
+let currentPlayer = 'X';
+let timer;
+let count = 6;
+let noClick = [];
+let index;
+let onePlayer = false;
+let compMove;
+
+let boxes = Array.from(document.getElementsByClassName("box"));  //array of all divs to make setting event listeners easier
+
+class Box {  //each box is an object with an element to point to, a value, and a clicked boolean
+    constructor(element, value) {
+        this.element = element;
+        this.value = value;
+        this.clicked = false;
+        this.xClick = false;
+        this.oClick = false;
+    }
+}
+let one = new Box(document.getElementById("one"), 1);
+let two = new Box(document.getElementById("two"), 2);
+let three = new Box(document.getElementById("three"), 3);
+let four = new Box(document.getElementById("four"), 4);
+let five = new Box(document.getElementById("five"), 5);
+let six = new Box(document.getElementById("six"), 6);
+let seven = new Box(document.getElementById("seven"), 7);
+let eight = new Box(document.getElementById("eight"), 8);
+let nine = new Box(document.getElementById("nine"), 9);
+
+boxArray = [one, two, three, four, five, six, seven, eight, nine];  //array of each object for referencing
+//various lookup objects for different outputs in the program
+let boxIDLookUp = {
+    'one': one,
+    'two': two,
+    'three': three,
+    'four': four,
+    'five': five,
+    'six': six,
+    'seven': seven,
+    'eight': eight,
+    'nine': nine
+}
+
+let numBoxLookUp = {
+    '1': one,
+    '2': two,
+    '3': three,
+    '4': four,
+    '5': five,
+    '6': six,
+    '7': seven,
+    '8': eight,
+    '9': nine
+}
```

Comment:
Since all your code is dependent on what you as the programmer have written (as apposed to some user input) so you can just decide what you're passing the boxes in as, and if you're consistent with your pointers throughout your code you wont need any lookup tables at all